synapseclient/models/schema_organization.py

```diff
@@ -0,0 +1,731 @@
+        self._fill_from_dict(response)
+
+    # Should this be named 'store_async'?
+    async def create_async(self, synapse_client: Optional["Synapse"] = None) -> None:
+        """
+        Creates this organization in Synapse
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor
+
+        Example:
+            from synapseclient.models import SchemaOrganization
+            from synapseclient import Synapse
+            import asyncio
+
+            syn = Synapse()
+            syn.login()
+
+            org = SchemaOrganization("my.org.name")
+            asyncio.run(org.create())
+
+        """
+        if self.id:
+            await self.get_async(synapse_client=synapse_client)
+        response = await create_organization(self.name, synapse_client=synapse_client)
+        self._fill_from_dict(response)
+
+    async def delete_async(self, synapse_client: Optional["Synapse"] = None) -> None:
+        """
+        Deletes this organization in Synapse
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor
+
+        Example:
+            from synapseclient.models import SchemaOrganization
+            from synapseclient import Synapse
+            import asyncio
+
+            syn = Synapse()
+            syn.login()
+
+            org = SchemaOrganization("my.org.name")
+            asyncio.run(org.delete())
+
+        """
+        if not self.id:
+            await self.get_async(synapse_client=synapse_client)
+        await delete_organization(self.id, synapse_client=synapse_client)
+
+    async def get_json_schema_list_async(
+        self, synapse_client: Optional["Synapse"] = None
+    ) -> list["JSONSchema"]:
+        """
+        Gets the list of JSON Schemas that are part of this organization
+
+        Returns: A list of JSONSchema objects
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor
+
+        Example:
+            from synapseclient.models import SchemaOrganization
+            from synapseclient import Synapse
+            import asyncio
+
+            syn = Synapse()
+            syn.login()
+
+            org = SchemaOrganization("my.org.name")
+            asyncio.run(org.get_json_schema_list_async())
+
+        """
+        response = list_json_schemas(self.name, synapse_client=synapse_client)
+        schemas = []
+        async for item in response:
+            schemas.append(JSONSchema.from_response(item))
+        return schemas
```

Comment:
This should also be a Generator too so that long responses aren't blocked for every result to come in before the results can be worked on

Reply:
What you would do here is to use this function:

https://github.com/Sage-Bionetworks/synapsePythonClient/blob/eb5f2e88fc3e779594e75484387b8304110a60fb/synapseclient/core/async_utils.py#L96-L152


How it's used looks like this:

https://github.com/Sage-Bionetworks/synapsePythonClient/blob/eb5f2e88fc3e779594e75484387b8304110a60fb/synapseclient/models/curation.py#L233-L255

You would use the `@skip_async_to_sync` decorator for the async method, and then manually implement the generator within the `organization.get_json_schema_list` method - It would internally call `organization.get_json_schema_list_async`.

fyi: 
It's a part of this open PR: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1253 - If you can review and approve, it'll be available for us to use.

Reply:
@BryanFauble This is #1253, do you mean a different PR?